### ISSUE TITLE
[NumberFieldActions] java.lang.NoSuchMethodException during Collection<Number> -> Number mapping field actions

### ISSUE THREAD

**Issue by mmelko** [atlasmap/atlasmap#368]
When  a transformation with defined `sourceCollectionType = CollectionType.ALL` is used (i.e Average) following exception is thrown during mapping:
```java
java.lang.NoSuchMethodException: io.atlasmap.actions.NumberFieldActions.average(io.atlasmap.v2.Action, java.lang.Number)
	at java.lang.Class.getMethod(Class.java:1786)
	at io.atlasmap.core.DefaultAtlasFieldActionService.processAction(DefaultAtlasFieldActionService.java:255) 
```

Average method from NumberFieldActions:
```java
average(Action action, Object input)
```

Mapping.xml: 
```xml
 <Mappings>
        <Mapping xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:type="Collection" collectionType="List" mappingType="Collection">
            <Mappings>
                <Mapping xsi:type="Mapping" id="mapping.648264" mappingType="Map">
                    <InputField xsi:type="ns2:JavaField" name="integers" docId="io.atlasmap.qe.test.SourceListsClass" path="/integers&lt;&gt;" fieldType="Integer"/>
                    <OutputField xsi:type="ns2:JavaField" name="targetDouble" docId="io.atlasmap.qe.test.TargetMappingTestClass" path="/targetDouble" fieldType="Double">
                        <Actions>
                            <Average/>
                        </Actions>
                    </OutputField>
                </Mapping>
            </Mappings>
        </Mapping>
    </Mappings>
```

**Comment by igarashitm** [comment]
Right, the field actions which requires a collection as source fields can't be utilized due to #338. Changed to enhancement as currently there's no way to specify a field action on a whole collection in UI.

**Comment by igarashitm** [comment]
Putting aside the #338, there still is a bug in UI, which allows user to choose "Average" field action on non-collection target field. As Average requires collection as source field, we shouldn't show it in the transformation list.

**Comment by jpav** [comment]
After realizing the real issue is with collection-based functions showing up when they shouldn't, unblocking this a treating as a bug in the UI.

**Comment by jpav** [comment]
Yes, I should be able to address this in that timeframe

**Comment by jpav** [comment]
Did you use the same mapping file, or were you still able to choose the average transformation from the UI?

**Comment by jpav** [comment]
I just tried mapping between an integer and double field, and I don't see any way to use the Add function. How are you creating this mapping? Was it an existing one or did you recreate after the fix for this?

**Comment by igarashitm** [comment]
@mmelko do you still see `Add` transformation in UI? `Add` is also the one only consume collection, so shouldn't appear for `Number` target field in UI.

**Comment by jpav** [comment]
Yes, that's why I asked my question. I'm not seeing it show up as an option, so it looks like things are working as expected to me.

**Comment by mmelko** [comment]
Well I have mapping from Collection<Number> to Number as I mentioned in tittle.

**Comment by jpav** [comment]
Is this an old mapping then? I don't know of a way via the UI to create a mapping from a source collection and add a transformation.

**Comment by mmelko** [comment]
No .. I'm always using UI for mapping creations. You can add transformation when you reveal "Edit mapping":
<img width="437" alt="screen shot 2018-04-18 at 17 10 48" src="https://user-images.githubusercontent.com/6814482/38940877-930ab482-432b-11e8-9ff3-38b3e971826a.png">

@igarashitm Transformations are showing up correctly only when collection is involved but they can't be used due to this issue.

**Comment by jpav** [comment]
Ah, I see how to do this.  @mmelko 's scenario is possible if you you an array as a source.

**Comment by mmelko** [comment]
@igarashitm I believe "Add" is meant to be as mathematical operation not adding into array.

**Comment by jpav** [comment]
@mmelko I is a mathematical operation, but it is meant to add the numbers from a source collection/array into a target number.

**Comment by mmelko** [comment]
@igarashitm @jpav  Ok so for Collection<Number> -> Number mappings there are no transformations which is correct (I guess) .. but still there is few other examples where I can select incorrect transformation. I'm creating separate issue for that. 
I still believe that root of this problem (NoSuchMethod) wasn't fixed though.

**Comment by igarashitm** [comment]
This happens if inappropriate field action is specified. I know the way of resolving field actions is pretty vulnerable, but it's where we are today.

**Comment by mmelko** [comment]
FYI https://github.com/atlasmap/atlasmap/issues/393
